src/store/ducks/opensrp/PlanDefinition/index.ts

```diff
@@ -166,3 +168,35 @@ export function getPlanDefinitionsArray(
   }
   return result;
 }
+
+/** RESELECT USAGE STARTS HERE */
+
+/** This interface represents the structure of plan definition filter options/params */
+export interface PlanDefinitionFilters {
+  title?: string /** plan object title */;
+}
+
+/** planDefinitionsArrayBaseSelector select an array of all plans
+ * @param state - the redux store
+ */
+export const planDefinitionsArrayBaseSelector = (planKey?: string) => (
+  state: Registry
+): PlanDefinition[] =>
+  values((state as any)[reducerName][planKey ? planKey : 'planDefinitionsById']);
```

Comment:
Do we still need to pass planKey as an argument here. It seems to me that it is not needed in this module. The reason it is required in `src/store/plans` ducks, is because the store state in that ducks has 2 top level properties, and thus one has to choose which of them they will use when creating the selector.